sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionEvaluatesToConstant.cs

```diff
@@ -96,6 +107,28 @@ private static void CheckForRedundantConditions(ExplodedGraph explodedGraph, Syn
             }
         }
 
+        private static bool IsLoopConditionInLoopWithBreak(ExpressionSyntax constantExpression)
+        {
+            var parentLoop = constantExpression.RemoveParentheses().Parent;
+
+            var loopStatement = (parentLoop as WhileStatementSyntax)?.Statement ??
+                (parentLoop as DoStatementSyntax)?.Statement ??
+                (parentLoop as ForStatementSyntax)?.Statement;
+
+            if (loopStatement == null)
+            {
+                return false;
+            }
+
+            var breakStatements = loopStatement.DescendantNodes().Where(n => n.IsAnyKind(LoopBreakingStatements));
+
+            var result = breakStatements
+                .Select(b => b.Ancestors().FirstOrDefault(a => a.IsAnyKind(LoopStatements)))
+                .Any(loop => loop.Equals(parentLoop));

```

Comment:
"loop" could be null, because there is FirstOrDefault.
Perhaps use Equals(loop, parentLoop)? Or if it cannot be null, use First() instead